eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/PaymentOnion.scala

```diff
@@ -331,6 +336,17 @@ object PaymentOnion {
           ).flatten
           Standard(TlvStream(tlvs))
         }
+
+        /** Create a standard trampoline inner payload instructing the trampoline node to wait for a trigger before sending an async payment. */
+        def createNodeRelayForAsyncPayment(amount: MilliSatoshi, expiry: CltvExpiry, nextNodeId: PublicKey, invoiceFeatures: Features[InvoiceFeature]): Standard = {
+          val tlvs = Seq(
+            Some(AmountToForward(amount)),
+            Some(OutgoingCltv(expiry)),
+            Some(OutgoingNodeId(nextNodeId)),
+            Some(AsyncPaymentFeatures(invoiceFeatures.toByteVector))
+          ).flatten
+          Standard(TlvStream(tlvs))
```

Comment:
You don't really need to use optional fields and then flatten them since they are all filled:

```suggestion
          Standard(TlvStream(AmountToForward(amount), OutgoingCltv(expiry), OutgoingNodeId(nextNodeId), AsyncPaymentFeatures(invoiceFeatures.toByteVector)))
```

Reply:
fixed in [dc69ae5](https://github.com/ACINQ/eclair/pull/2435/commits/dc69ae5e372aba64a2795b7e25a20b620e82e60d)